JDK-8257547: Handle multiple prereqs on the same line in deps files

erikj79 opened the pull request:
After fixing JDK-8256810 and starting to look into backporting it, I discovered more potential failing cases. Certain versions of GCC may sometimes output multiple prerequisite files on the same line. I think the easiest way to handle this new issue is to split such lines.

When splitting lines, we need to make sure to not just split on any space. Some compilers output the : of the rule with a leading space, and already split lines will have a space before the backslash.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257547](https://bugs.openjdk.java.net/browse/JDK-8257547): Handle multiple prereqs on the same line in deps files


### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - **Reviewer**)
 * [Tim Bell](https://openjdk.java.net/census#tbell) (@tbell29552 - **Reviewer**) ⚠️ Review applies to 0c854066af2e12ab111a985c658dc32d0caede02


### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1548/head:pull/1548`
`$ git checkout pull/1548`

tbell29552 reviewed (approved):
As @magicus wrote - this is subtle but looks OKAY.

erikj79 commented:
I added a test as suggested. This uncovered some portability issues as the test failed on Macosx. I have modified the fix-deps-file macro even more to make it portable for all the versions of sed we are currently using.

tbell29552 commented on test/make/TestFixDepsFile.gmk:
```diff
@@ -0,0 +1,66 @@
+	$(ECHO) " $(WORKSPACE_ROOT)/bar \\" >> $(DEPS_FILE).expected
+	$(ECHO) " $(WORKSPACE_ROOT)/bar/baz \\" >> $(DEPS_FILE).expected
+	$(ECHO) " /foo/baz" >> $(DEPS_FILE).expected
+	$(DIFF) $(DEPS_FILE).expected $(DEPS_FILE)
```
Does this need to be:
$(DIFF) $(DEPS_FILE).expected $(DEPS_FILE).tmp

erikj79 replied:
No, the fix-deps-file macro takes file.tmp as input and outputs into file, $(DEPS_FILE) is the output file from the macro in this case.

tbell29552 replied:
OK

erikj79 commented:
/integrate
